services/secret-service/doc/openapi.yaml

```diff
@@ -299,22 +339,87 @@ components:
         - clientSecret
         - refreshTokenUrl
       properties:
-        name:
-          type: string
         clientId:
           type: string
         clientSecret:
           type: string
-        refreshTokenUrl:
+        redirectUri:
           type: string
+        endpoint:
+          type: object
+          properties:
+            auth:
+              type: string
+            token:
+              type: string
+            userinfo:
+              type: string
+            revocation:
+              type: string
+            endSession:
+              type: string
+          required:
+            - auth
+            - token
+        predefinedScope:
+          type: string
+        mappings:
+          type: object
+          properties:
+            scope:
+              type: object
+              properties:
+                key:
+                  type: string
+              required:
+                - key
+            externalId:
+              type: object
+              properties:
+                source:
+                  type: string
+                  enum: [id_token, access_token, userinfo]
+                key:
+                  type: string
+              required:
+                - source
+                - key
+      example: {
+        "_id": "5c06ee51120e789235d9dc19",
+        "endpoint": {
+          "auth": "https://accounts.google.com/o/oauth2/v2/auth?prompt=consent&scope={{scope}}&access_type=offline&include_granted_scopes=true&state={{state}}&redirect_uri={{redirectUri}}&response_type=code&client_id={{clientId}}",
+          "token": "https://www.googleapis.com/oauth2/v4/token",
+          "userinfo": "https://www.googleapis.com/oauth2/v4/userinfo"
+        },
+        "clientId": "myclientid",
+        "clientSecret": "shhhhhhhh",
+        "redirectUri": "http://localhost:9001/api/v1/callback",
+        "mappings": {
```

Comment:
Why do we need this?

Reply:
Which line exactly? If you mean `mappings` then it is optional but we actually need it to find or fetch the user reference. We are still working on it to minimize the API footprint.
You can ignore it, if you do not need it.

Reply:
An implementation may return more properties in the response but do we really need to expose them in the spec? You would force us to return them as well event though they make sense for your specific implementation only.

Reply:
It's ok with us removing the mappings from the spec.